torch/prototypes/nestedtensor/nested.py

```diff
@@ -0,0 +1,124 @@
+import torch
+
+
+def is_nested_tensor(obj):
+    return isinstance(obj, NestedTensor)
+
+
+def make_nested_tensor(obj):
+    if is_nested_tensor(obj):
+        return obj.clone().detach()
+    elif torch.is_tensor(obj):
+        return obj.clone().detach()
+    else:
+        assert isinstance(obj, list)
+        if len(obj) == 0:
+            return NestedTensor([])
+        for obj_ in obj:
+            assert(torch.is_tensor(obj_))
+        tensors = []
+        for obj_ in obj:
+            tensors.append(obj_.clone().detach())
+        return NestedTensor(tensors)
+
+
+class NestedTensor():
+    def __init__(self, tensors):
```

Comment:
Is this a private constructor in general?

Reply:
Yes, it's not supposed to be called directly akin to how you're not supposed to initialize torch.Tensor directly